.github/workflows/ci_test.yml

```diff
@@ -0,0 +1,133 @@
+name: e_e CI targets
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  release:
+    types:
+      - published
+
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install non-python dependencies on mac
+      if: runner.os == 'macOS'
+      run: |
+        brew install gcc
+        brew install automake suite-sparse
+        curl -sSL https://raw.githubusercontent.com/vallis/libstempo/master/install_tempo2.sh | sh
+    - name: Install non-python dependencies on linux
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get install libsuitesparse-dev
+        curl -sSL https://raw.githubusercontent.com/vallis/libstempo/master/install_tempo2.sh | sh
+    - name: Install dependencies and package
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install flake8 pytest black pytest-cov Cython numpy scipy
```

Comment:
Actually I see now. Its best to put these in a `pyproject.toml`. Something like [this](https://github.com/scikit-sparse/scikit-sparse/blob/master/pyproject.toml#L23). That way these are not actual dependencies and will get automatically installed when running `pip install`

Reply:
Yeah, they are needed for the Cython part of the setup, as you saw. The `*.toml` is slick. Is that just seen automatically by GitHub, like the `*.yml` files? 

Also, how did you figure out which numpy versions to pin to which Python versions? I think I should probably do the same thing for scipy.

Reply:
The `pyproject.toml` is part of the build process for pip so `pip install` looks for it.  For the numpy versions I based it on [this](https://github.com/scipy/oldest-supported-numpy) but you could probably use that directly like [this](https://github.com/healpy/healpy/blob/main/pyproject.toml#L7). 

I don't think you need `scipy` at all as a build dependency. Its not used in the `setup.py`. The only reason to do numpy like this is because the C-extensions use the numpy ABI which is only *forward* compatible, so it has to be build with the oldest supported version of numpy. This does not mean that the numpy dependency installed with the package needs to be the same (as long as the c-extensions are built with anversion <= the installed version). Does that make sense?

Reply:
Ok I got a bit obsesses with this. I think I got it working [here](https://github.com/jellis18/enterprise_extensions/pull/1). The main problem was that the `libiomp` is a MKL thing and is not generally available except through conda. My solution is somewhat hacky in that you have to use an environment variable. We would need to automatically detect if this library is installed but at least I think the build works.

Reply:
Awesome! Thanks. I merged those changes into my master, so they are now in the PR.

Reply:
@jellis18, did you remove the `*.c` files form the outlier module because they weren't needed, or  for some other reason?

Reply:
They aren't needed. They get generated from the cython code.